h/schemas/forms/accounts/login.py

```diff
@@ -9,14 +9,27 @@
 _ = i18n.TranslationString
 
 
+@colander.deferred
+def _deferred_username_widget(node, kw):
+    """Return a username widget that autofocuses if username isn't pre-filled."""
+    return deform.widget.TextInputWidget(autofocus=_should_autofocus_username(kw))
+
+
+@colander.deferred
+def _deferred_password_widget(node, kw):
+    """Return a password widget that autofocuses if username *is* pre-filled."""
+    return deform.widget.PasswordWidget(autofocus=not _should_autofocus_username(kw))
```

Comment:
These have to be at the top of the file 'cos they're used in class-level declarations. (Normally we'd hide private helpers like this away at the bottom of the file.)

Reply:
I didn't actually add tests for when `autofocus` is `True` or `False` for the username and password widgets because I don't know how to test that. Maybe there's a nice way to unit test it, maybe not. Would require more knowledge of Colander and/or Deform. I'm gonna opt not to bother